setuptools/command/build_py.py

```diff
@@ -302,6 +302,19 @@ def _get_platform_patterns(spec, package, src_dir):
             for pattern in raw_patterns
         )
 
+    def get_source_files(self):
+        py_files = [module[-1] for module in self.find_all_modules()]
+
+        possible_stub_files = set(os.path.splitext(f)[0] + ".pyi" for f in py_files)
+        stub_files = [f for f in possible_stub_files if os.path.isfile(f)]
+
+        possible_py_typed_files = set(
+            os.path.join(os.path.dirname(f), "py.typed") for f in py_files
```

Comment:
I am curious, is there any particular reason why you went with `py_files` => `os.path.dirname`, instead of iterating over `self.packages` and using `self.get_package_dir`?

Can a wheel have a `py.typed` file at the root (without being inside of a package folder)?

Reply:
There wasn't any particular reason, I was just trying to get something to work before trying to refine it (I should have said this in the pull request). I'll remember to check back on this later but for now I'll leave it as is in case the code needs to be drastically changed to address the issue of files not being included in the wheel.

I tried reading https://peps.python.org/pep-0561/#packaging-type-information but couldn't tell if it addresses your question about the `py.typed` file. To me, it slightly sounds like it says that the file must be in a package folder but I don't think I fully understand what it says at the moment.

Reply:
No problems, you can refine that at a later stage, that is completely fine.

I also have the impression that you cannot have `py.typed` outside of a package folder. Upon further read of the link you shared, I think there is evidence of that in the following paragraphs:

> This PEP does not support distributing typing information as part of module-only distributions or single-file modules within namespace packages.
> The single-file module should be refactored into a package and indicate that the package supports typing as described above.

It seems that you cannot have a top-level `.pyi`, it should always be contained inside a package dir.